### PR TITLE
Adding AzureAuthentication.rst

### PR DESCRIPTION
**Pull request by ianhelle**
Updating several docs to bring authentication details up-to-date and adding cross-references to the AzureAuthentication doc. Fixing a few formatting issues in docs.

### CODE REVIEW

File: docs/source/data_acquisition/AzureBlobStorage.rst [hunk 1]
```diff
@@ -6,9 +6,10 @@ Description
 
 Azure Blob Storage provides a simple and flexible way to store and access data of any kind.
 This makes it ideal for storing a range of data relating to security investigations, whether
-it be raw data to analyse or to store outputs and findings.
+it be raw data to analyze or to store outputs and findings.
```
Comment (petebryan) [inline docs/source/data_acquisition/AzureBlobStorage.rst:9]: You've been in the US too long ;)

File: docs/source/getting_started/AzureAuthentication.rst [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+- Device Code
+
+This means that MSTICPy will try to obtain credentials of each these
+types in turn. If it finds a matching credential type it will try
+to use that to authenticate you to the Azure service that you are
+trying to access. In the default case it will do the following:
+
+1. Check to see if you are running on a system (a host/virtual machine)
+   that has a system-assigned identity OR (as in the case of Azure
+   Machine Learning) has support for user-assigned managed identities.
```
Comment (petebryan) [inline docs/source/getting_started/AzureAuthentication.rst:78]: Do we want to link to documents on Azure Managed Identities here? Its not something people are often familiar with?

**Review by petebryan — APPROVED** [review]

File: docs/source/data_acquisition/AzureBlobStorage.rst (hunk 1 above)
Comment (ianhelle) [reply docs/source/data_acquisition/AzureBlobStorage.rst:9]: haha
Yeah - just following the wavy lines :-)

**Comment by ianhelle** [comment]
Document-only PR - merging without build/re-approval

**Pull request merged**